pkg/lock/simple_redis.go

```diff
@@ -0,0 +1,134 @@
+package lock
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/utils"
+)
+
+const luaRefresh = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("pexpire", KEYS[1], ARGV[2]) else return 0 end`
+const luaRelease = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("del", KEYS[1]) else return 0 end`
+
+type fakeRWLock struct {
+	ErrorLocker
+}
+
+func (l fakeRWLock) Lock() error    { return l.ErrorLocker.Lock() }
+func (l fakeRWLock) RLock() error   { return l.ErrorLocker.Lock() }
```

Comment:
It may be dangerous to have `RLock` behave like `Lock` since calling `RLock` recursively will deadlock while it is a perfectly valid use of a `RWLock`. I'm not sure that this would not cause a deadlock in our current code, it would require further analysis and tests.

One option would be to have `RLock` be a noop. The lock would prevent corruption of the FS by locking the writes but would not assure the reads to be fail-safe, which may be sufficient in our case.